cmd/netpolicy/cmd/command_test.go

```diff
@@ -2,159 +2,162 @@ package cmd
+	return res
 }
 
-// check that expected is contained in actual
-func containedOutput(t *testing.T, actual, expected, testName string, isFile bool) {
-	isContained := strings.Contains(actual, expected)
-	assert.True(t, isContained, "test %s error: %s not contained in %s, isFile: %d", testName, expected, actual, isFile)
+// build a new command with args list and execute it, returns the actual output from stdout and the execute err if exists
+func buildAndExecuteCommand(args []string) (string, error) {
+	preTestRun()
+	cmd := newCommandRoot()
+	cmd.SetArgs(args)
+	err := cmd.Execute()
+	actualOut := postTestRun()
+	return actualOut, err
 }
 
-func clean(test cmdTest) {
-	if test.hasFile {
-		os.Remove(outFileName)
+// append the optional args of a command if the values are not empty
+func addCmdOptionalArgs(format, outputFile, focusWorkload string) []string {
+	res := []string{}
+	if focusWorkload != "" {
+		res = append(res, "--focusworkload", focusWorkload)
+	}
+	if format != "" {
+		res = append(res, "--output", format)
 	}
+	if outputFile != "" {
+		res = append(res, "-f", outputFile)
+	}
+	return res
 }
 
-func runTest(test cmdTest, t *testing.T) {
-	// run the test and get its output
-	preTestRun()
-	rootCmd := newCommandRoot()
-	rootCmd.SetArgs(test.args)
-	err := rootCmd.Execute()
-	if !test.isErr {
-		require.Nilf(t, err, "expected no errors, but got %v", err)
-	} else {
-		require.NotNil(t, err, "expected error, but got no error")
+// compares actual vs expected output
+func compareActualVsExpectedOutput(t *testing.T, dir, testName, expectedOutputFileName, actualOutput, outputFile string) {
+	expectedOutputFile := filepath.Join(getTestsDir(), dir, expectedOutputFileName)
+	expectedOutput, err := os.ReadFile(expectedOutputFile)
+	require.Nil(t, err)
+	actualOutputFileName := "actual_" + expectedOutputFileName
+	actualOutputFile := filepath.Join(getTestsDir(), dir, actualOutputFileName)
+	if string(expectedOutput) != actualOutput {
+		// generate actual file for self check
+		err := writeBufToFile(actualOutputFile, []byte(actualOutput))
+		require.Nil(t, err, "test: %q", testName)
 	}
-	actual := postTestRun(test.isErr)
-
-	// check if has file and if it exists
-	var fileContent []byte
-	if test.hasFile {
-		_, err := os.Stat(outFileName)
-		require.Nil(t, err)
-		fileContent, err = os.ReadFile(outFileName)
-		require.Nil(t, err)
+	require.Equal(t, string(expectedOutput), actualOutput,
+		"output mismatch for test %q, actual output file %q vs expected output file: %q",
+		testName, actualOutputFile, expectedOutputFile)
+	if outputFile != "" {
+		_, err := os.Stat(outputFile)
+		require.Nil(t, err, "test: %q", testName)
+		fileContent, err := os.ReadFile(outputFile)
+		require.Nil(t, err, "test: %q", testName)
+		require.Equal(t, string(expectedOutput), string(fileContent),
+			"output mismatch for test %q, actual output file %q vs expected output file: %q", testName, outputFile, expectedOutputFile)
```

Comment:
is this function similar to the one from `pkg/internal/testutils`? can the code duplication be avoided?

Reply:
yes both are very similar (only the part of `if outputFile != ""` is relevant for here only)
I think we can avoid code duplications by moving `internal/testutils` to be sibling of both `cmd` and `pkg`. Is this possible?
(this also true for `getTestsDir`)

Reply:
Why not moving `pkg/internal/testutils` to be  `pkg/testutils`, and export this function, so that it can also be used under `cmd` tests?

Reply:
sure we can. (I thought `testutils ` was intended to be internal only)

Reply:
done